src/core/Wyam.Common/Util/Splitter.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Wyam.Common.Util
+{
+    /// <summary>
+    /// Splits a string based on a delimiter.
+    /// </summary>
+    /// <remarks>
+    /// This class is used to split some content on a deliimiter.
+    /// It is extracted from the FrontMatter module.
+    /// </remarks>
+    public class Splitter
+    {
+        private readonly string _delimiter;
+        private readonly bool _repeated;
+        private bool _ignoreDelimiterOnFirstLine = true;
+
+        /// <summary>
+        /// Constructs a Splitter instance with the specified delimiter.
+        /// </summary>
+        /// <param name="delimiter">The delimiter to use.</param>
+        /// <param name="repeated">True if the delimiter will be repeated. Defaults to false.</param>
+        public Splitter(string delimiter, bool repeated = false)
+        {
+            _delimiter = delimiter;
+            _repeated = repeated;
+        }
+
+        /// <summary>
+        /// Ignores the delimiter if it appears on the first line. This is useful when processing Jekyll style front matter that
+        /// has the delimiter both above and below the front matter content. The default behavior is <c>true</c>.
+        /// </summary>
+        /// <param name="ignore">If set to <c>true</c>, ignore the delimiter if it appears on the first line.</param>
+        /// <returns>The current Splitter instance.</returns>
+        public Splitter IgnoreDelimiterOnFirstLine(bool ignore = true)
+        {
+            _ignoreDelimiterOnFirstLine = ignore;
+            return this;
+        }
+
+        /// <summary>
+        /// Split the input.
+        /// </summary>
+        /// <param name="input">The input to split.</param>
+        /// <returns>A list of the split sections.</returns>
+        public List<string> Split(string input)
```

Comment:
Wondering if we should return something other than a `List<string>` since it's always going to have either 1 or 2 items. 

One option would be to return the front matter content (or `null` if there isn't any) and then use a `ref` argument for the input and post-split content, that way if we didn't find the delimiter we could just leave the ref untouched: `string Split(ref string content) { ... }`

Another option would be to return a tuple: `(string frontMatter, string content) Split(string input) { ... }`

Thoughts?